### PR TITLE
nixos: make services.smartd much more helpful

### PR DESCRIPTION
**Pull request by oxij**
Now it generates notifications for auto-detected devices as well as
for explicitly configured ones, sends well formed e-mails and supports
immediate `wall` and `xmessage` notifications.

### CODE REVIEW

**Comment by bjornfor** [comment]
I didn't test it but it looks good to me.

File: nixos/modules/services/monitoring/smartd.nix [hunk 1]
```diff
@@ -64,36 +99,116 @@ in
...
+
+          Set to false to monitor the devices listed in
+          <option>services.smartd.devices</option> only.
         '';
       };
 
+      notifications = {
+
+        mail = {
+          enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/monitoring/smartd.nix]: Please use `mkEnableOption`.

File: nixos/modules/services/monitoring/smartd.nix [hunk 2]
```diff
@@ -64,36 +99,116 @@ in
...
+
+              You should probably enable
+              <option>services.postfix</option> or some other MDA for
+              this to work.
+            '';
+          };
+        };
+
+        wall = {
+          enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/monitoring/smartd.nix]: Please use `mkEnableOption`.

File: nixos/modules/services/monitoring/smartd.nix [hunk 3]
```diff
@@ -64,36 +99,116 @@ in
...
+            default = true;
+            type = types.bool;
+            description = "Whenever to send wall notifications to all users.";
+          };
+        };
+
+        x11 = {
+          enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/monitoring/smartd.nix]: Please use `mkEnableOption`.

File: nixos/modules/services/monitoring/smartd.nix [hunk 4]
```diff
@@ -64,36 +99,116 @@ in
...
+
+          display = mkOption {
+            default = ":${toString config.services.xserver.display}";
+            type = types.string;
+            description = "DISPLAY to send X11 notifications to.";
+          };
+        };
+
+        test = mkOption {
+          default = false;
```
Comment (jagajaga) [inline nixos/modules/services/monitoring/smartd.nix]: Please use `mkEnableOption`.

**Comment by oxij** [comment]
@jagajaga I changed one of them to `mkEnableOption`, others won't do because they have nontrivial defaults.

**Comment by jagajaga** [comment]
My fault, haven't noticed that.

**Comment by jagajaga** [comment]
:+1: on merging.

**Pull request merged**